tests/src/system/basic/WskRuleTests.scala

```diff
@@ -0,0 +1,297 @@
+        (wp, assetHelper) =>
+            val ruleName = "r1to1"
+            val triggerName = "t1to1"
+            val actionName = "a1 to 1"
+
+            ruleSetup(Seq(
+                (ruleName, triggerName, (actionName, defaultAction))),
+                assetHelper)
+
+            val now = Instant.now
+            val run = wsk.trigger.fire(triggerName, Map("payload" -> testString.toJson))
+
+            withActivation(wsk.activation, run) {
+                triggerActivation =>
+                    triggerActivation.cause shouldBe None
+
+                    withActivationsFromEntity(wsk.activation, ruleName, since = Some(Instant.ofEpochMilli(triggerActivation.start))) {
+                        _.head.cause shouldBe Some(triggerActivation.activationId)
+                    }
+
+                    withActivationsFromEntity(wsk.activation, actionName, since = Some(Instant.ofEpochMilli(triggerActivation.start))) {
+                        _.head.response.result shouldBe Some(testResult)
+                    }
+            }
+    }
+
+    it should "not fire an action if the rule is deleted when the trigger is fired" in withAssetCleaner(wskprops) {
+        (wp, assetHelper) =>
+            val ruleName = "ruleDelete"
+            val triggerName = "ruleDeleteTrigger"
+            val actionName = "ruleDeleteAction"
+
+            assetHelper.withCleaner(wsk.trigger, triggerName) {
+                (trigger, name) => trigger.create(name)
+            }
+            assetHelper.withCleaner(wsk.action, actionName) {
+                (action, name) => action.create(name, Some(defaultAction))
+            }
+            assetHelper.withCleaner(wsk.rule, ruleName, false) {
+                (rule, name) => rule.create(name, triggerName, actionName)
+            }
+
+            val first = wsk.trigger.fire(triggerName, Map("payload" -> "bogus".toJson))
+            wsk.rule.delete(ruleName)
+            wsk.trigger.fire(triggerName, Map("payload" -> "bogus2".toJson))
+
+            withActivation(wsk.activation, first) {
+                activation =>
+                    // tries to find 2 activations for the action, should only find 1
+                    val activations = wsk.activation.pollFor(2, Some(actionName), since = Some(Instant.ofEpochMilli(activation.start)), retries = 5)
+
+                    activations.length shouldBe 1
+            }
+    }
+
+    it should "not fire an action if the rule disbaled when the trigger is fired, being able to reenable it" in withAssetCleaner(wskprops) {
+        (wp, assetHelper) =>
+            val ruleName = "ruleDisable"
+            val triggerName = "ruleDisableTrigger"
+            val actionName = "ruleDisableAction"
+
+            ruleSetup(Seq(
+                (ruleName, triggerName, (actionName, defaultAction))),
+                assetHelper)
+
+            val first = wsk.trigger.fire(triggerName, Map("payload" -> testString.toJson))
```

Comment:
using the same testString makes me a little nervous - it's not as bad as it might seem given the `since` in the poll method but could be weaker than using unique time stamps.